app/features/send/cashu-send-quote-hooks.ts

```diff
@@ -0,0 +1,500 @@
+            );
+          }
+          const existingQuotesForMint = acc[account.mintUrl] ?? [];
+          acc[account.mintUrl] = existingQuotesForMint.concat(quote.quoteId);
+          return acc;
+        },
+        {},
+      );
+
+      const subscriptions = Object.entries(quotesByMint).map(
+        ([mintUrl, quotes]) => {
+          const wallet = getCashuWallet(mintUrl);
+          return wallet.onMeltQuoteUpdates(
+            quotes,
+            handleMeltQuoteUpdate,
+            // TODO: what should we do here on error?
+            (error) =>
+              console.error('Melt quote updates socket error', {
+                cause: error,
+              }),
+          );
+        },
+      );
+
+      const subscriptionCancellers = await Promise.all(subscriptions);
+
+      return () => {
+        subscriptionCancellers.forEach((cancel) => cancel());
+      };
+    };
+
+    subcribeToMeltQuoteUpdates(sendQuotes);
+  }, [sendQuotes, accountsCache, handleMeltQuoteUpdate]);
+
+  useEffect(() => {
+    // We need to check the state of the quote upon expiration because there is no state change for the expiration
+    // so socket will not notify us.
+    if (sendQuotes.length === 0) return;
+
+    const timeouts: LongTimeout[] = [];
+
+    for (const sendQuote of sendQuotes) {
+      const msUntilExpiration =
+        new Date(sendQuote.expiresAt).getTime() - Date.now();
+      const quoteTimeout = setLongTimeout(async () => {
+        const account = await accountsCache.getLatest(sendQuote.accountId);
+        if (!account || account.type !== 'cashu') {
+          throw new Error(`Account not found for id: ${sendQuote.accountId}`);
+        }
+
+        const wallet = getCashuWallet(account.mintUrl);
+        const meltQuote = await wallet.checkMeltQuote(sendQuote.quoteId);
+
+        return handleMeltQuoteUpdate(meltQuote);
+      }, msUntilExpiration);
+      timeouts.push(quoteTimeout);
+    }
+
+    return () => {
+      timeouts.forEach((timeout) => clearLongTimeout(timeout));
+    };
+  }, [sendQuotes, handleMeltQuoteUpdate, accountsCache]);
+}
+
+export function useTrackUnresolvedCashuSendQuotes() {
+  const cashuSendService = useCashuSendQuoteService();
+  const unresolvedSendQuotes = useUnresolvedCashuSendQuotes();
+
+  useOnMeltQuoteStateChange({
+    sendQuotes: unresolvedSendQuotes,
+    onUnpaid: (account, send, meltQuote) => {
+      // In case of failed payment the mint will flip the state of the melt quote back to UNPAID.
+      // In that case we don't want to initiate the send again so we are only initiating the send if our quote state is also UNPAID which won't be the case if the send was already initiated.
+      if (send.state === 'UNPAID') {
+        // TODO: this should probaby trigger mutation that will then call triggerSend. That way mutation will be responsible for errors and retries.
+        cashuSendService
+          .initiateSend(account, send, meltQuote)
+          .catch((error) => {
```

Comment:
this was added to handle errors. btw note how I used CashuErrorCodes. I think in some of your recent PRs you forgot that and used codes directly so just a reminder